src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -196,3 +212,68 @@ def _update_job_config(cls):
             cls._dispatcher = _StandaloneJobDispatcher()
         else:
             cls._dispatcher = _DevelopmentJobDispatcher()
+
+    @classmethod
+    def cancel_job(cls, job: Job):
+        to_cancel_jobs = set()
+        to_cancel_jobs.update([job])
+        to_cancel_jobs.update(cls.__find_subsequent_jobs(job.submit_id, set(job.task.output.keys())))
+        with cls.lock:
+            cls.__remove_blocked_jobs(to_cancel_jobs)
+            cls.__remove_jobs_to_run(to_cancel_jobs)
+            cls.__cancel_processes(to_cancel_jobs)
+            cls.__unlock_edit_on_outputs(to_cancel_jobs)
+        # if not cls.jobs_to_run.empty():
+        #     cls.__execute_jobs()
+        # else:
+        #     cls.__unblock_jobs()
+        #     cls.__execute_jobs()
```

Comment:
I commented this part of the code 'cause I believe it is not necessary.

Reply:
@jrobinAV I would say in the scenario where the process pool is full, and we are able to cancel 1 process, we will need to call `__execute_jobs()` to execute the next job since the previous canceled job was never finished and it can't initiate the next job. Of course, currently, this is not needed as we can't cancel a process yet